lesson01.md

```diff
@@ -12,5 +21,587 @@ Lesson 1, Tuesday, 2021-xx-xx
+
+---
+
+### Data Types: Number
+
+* One basic data type is called **Number**
+* A Number can be a positive *integer* `(1, 2, 42)`
+* or a negative one `(-1, -2, -42)`
+* or a *floating point* `(0.25, -4.5)`
+
+---
+
+### DIY
+
+Open the console, and type in the following numbers:
+
+* Your age
+* The number of siblings you have
+
+
+---
+
+
+### Numerical Operators
+
+Basic mathematical operators:
+
+`+` addition <br>
+`-` subtraction <br>
+`*` multiplication <br>
+`/` division <br>
+`**` exponentiation
+
+---
+
+### Quick Quiz
+
+Let's say, everyone here wants to go to the cinema with their siblings.
+
+How many people in total are going to the cinema?
+<!-- .element: class="fragment" -->
+
+A ticket to watch the movie *The Joker* costs 8€, how much do we have to pay in total?
+<!-- .element: class="fragment" -->
+
+Harald and Adrian volunteered to pay for the tickets, how much each one pays?
+<!-- .element: class="fragment" -->
+
+Can they afford buying popcorn too?
+<!-- .element: class="fragment" -->
+
+---
+
+### Data Type: String
+
+* A **String** is a sequence of characters 
+    * a good way to represent text
+* It starts and ends with a `"` double quote
+* It can also start and end with a `'` single quote
+* Examples: `"Hello"`, `'Sevtap'`
+
+---
+
+### DIY
+
+Open the console, and type in the following strings:
+
+* Your full name
+* The name of your neighbor on the right
+
+---
+
+### Strings and Quotes
+
+Strings start and end with  (`'`) or (`"`). But what if we want to add a quote within our string?
+
+```js
+"He said: "Hello""
+```
+
+
+We can escape the quote with a backslash (`\`):
+
+```js
+"He said: \"Hello\""
+```
+
+---
+
+### Practice
+
+Can you do the same using `'` single quotes?
+
+Try writing this string:
+
+`'She said: 'Hello''`
+
+---
+
+### Special Characters
+
+If you want a backslash in a string, you need to escape it: `"\\"`
+
+There are some special characters in strings, for example:
+
+* `"\n"` - new line
+* `"\t"` - tab
+
+
+---
+
+
+### Quiz: Which strings are correct?
+
+```plaintext
+"Hello"
+'World'
+"He said "hello" to me"
+'Let's go!'
+""
+'This is a\nnew line'
+'This is a backslash: \'
+```
+
+---
+
+### Solution
+
+```js
+"Hello"                        // CORRECT
+'World'                        // CORRECT
+"He said "hello" to me";       // WRONG - unescaped quotes
+'Let's go!'                    // WRONG - unescaped '
+""                             // CORRECT
+'This is a\nnew line';         // CORRECT
+'This is a backslash: \';      // WRONG - escaped '
+```
+
+---
+
+The addition operator can be used on Strings and Numbers:
+
+```js
+1 + 1                // is 2
+"hello" + " world"   // is "hello world"
+```
+
+---
+
+### Importance of data types
+
+```js
+1 + 1        // is 2
+"1" + "1"    // is "11"
+```
+
+Make sure to always choose the correct data type for your data,
+because the operators in JavaScript behave differently depending on the data type.
```

Comment:
```suggestion
Make sure to always choose the correct data type for your data,
since the operators in JavaScript behave differently depending on the data type.
```